### ISSUE TITLE
About the problem of abnormal_sequences

### ISSUE THREAD

**Issue by ljy1234-special** [ristea/aed-mae#2]
Hello! The original UBnormal dataset does not provide an `abnormal_sequences` files in the UBnormal dataset, so I am wondering how you got this file, or could you provide a download link?
Thank you very much.

**Comment by zw-absin** [comment]
This one seems to be self-generating in ./util/create_anomalies/create_abnormal_sequences.py, line 16. By the way, to you know how to get the frames_abnormal and masks_abnormal in train_dataset.py line 46-48? I didn't find these in the shanghaitech dataset downloaded by myself.

**Comment by ljy1234-special** [comment]
Thank you! I get you.I think the frames_abnormal and masks_abnormal directories seem to be generated in ./util/create_anomalies/integrate_abnormal_objects.py(frames and masks).By the way, do you know the groundtruth(*.txt) about avenue and shanghaitech dataset in ./data/test_dataset.py line 40 which is corresponding to path in the ./configs/configs.py?

**Comment by zw-absin** [comment]
> Thank you! I get you.I think the frames_abnormal and masks_abnormal directories seem to be generated in ./util/create_anomalies/integrate_abnormal_objects.py(frames and masks).By the way, do you know the groundtruth(*.txt) about avenue and shanghaitech dataset in ./data/test_dataset.py line 40 which is corresponding to path in the ./configs/configs.py?

Thanks. I'll check it later. If I understand right, I asked this question in 'https://github.com/ristea/aed-mae/issues/3'. No one answered. For shanghaitech, It provides pixel_mask & frame_mask(.npy) in the test set. So I jast replaced .txt with .npy in test_dateset.py line 40. But I'm not sure which one is used in 'test_datset.py'.         The instantiation of the class(AbnormalDatasetGradientsTest) is fed directly into torch.utils.data.DataLoader in main.py line 41. If this helps determine the labels(test_dataset.py line 43) content?

**Comment by ljy1234-special** [comment]
Yeah！You are right! It is indeed the question in [#3](https://github.com/ristea/aed-mae/issues/3). Do you know the groundtruth corresponding to each video in the `Avenue dataset`? It contains a file similar to `vol01.mat`, but the file data doesn't match the video frames, and I don't know what this file represents?

**Comment by 1s3ss** [comment]
Hello, I would like to ask how to get the annotations on line 14 of "./util/create_anomalies/create_abnormal_sequences.py" ?
 I downloaded the avenue dataset and there are no annotations in it. Thank you so much

**Comment by ljy1234-special** [comment]
The annotation on line 14 of `. /util/create_anomalies/create_abnormal_sequences.py` the comment on line 14 is an abnormal that exists in the **UBnormal** dataset, not in Avenue. @1s3ss 

> Hello, I would like to ask how to get the annotations on line 14 of "./util/create_anomalies/create_abnormal_sequences.py" ? I downloaded the avenue dataset and there are no annotations in it. Thank you so much

**Comment by 1s3ss** [comment]
> The annotation on line 14 of the comment on line 14 is an abnormal that exists in the **UBnormal** dataset, not in Avenue. @1s3ss`. /util/create_anomalies/create_abnormal_sequences.py`
> 
> > Hello, I would like to ask how to get the annotations on line 14 of "./util/create_anomalies/create_abnormal_sequences.py" ? I downloaded the avenue dataset and there are no annotations in it. Thank you so much

I get you, thank you very much!

**Comment by CroitoruAlin** [comment]
Hello!
1.Regarding the ground-truth anomaly annotations, we uploaded the files that we used in the drive folder.
2.The UBNormal dataset is a special case because it already contains anomalies in the training subset. Thus, we didn't add synthetic anomalies for this dataset

**Comment by JPerAsperaadAstra** [comment]
Hi, thanks for open-source this outstanding work! A question about anomaly sequence generating: Should we first extract UBnormal dataset's frames since in line 13 of create_abnormal_sequences.py we need this for line 21? Also, are the train val test dataset split the same as those in UBnormal?
